### PR TITLE
Add an action to validate generated code

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

This ensures that the generated code is fresh.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/validate-generated-code.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  my_first_job:
+    steps:
+      - name: My first step
+        uses: docker://gcr.io/cloud-builders/gradle
```
Comment (khanhtc1202) [inline .github/workflows/validate-generated-code.yaml]: do we need this?

Comment (nghialv) [reply .github/workflows/validate-generated-code.yaml]: haha. You are faster than me.

Comment (khanhtc1202) [reply .github/workflows/validate-generated-code.yaml]: 🤣

File: .github/workflows/validate-generated-code.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  analyze:
+    name: Analyze
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Generate code
+      uses: docker://gcr.io/pipecd/codegen:0.2.0
+      with:
+        args: /github/workspace
+     - name: Show Git status
```
Comment (khanhtc1202) [inline .github/workflows/validate-generated-code.yaml:17]: indent error?

**Comment by khanhtc1202** [comment]
Here you go 🚀 
/approve

**Pull request merged**